projects/ipdg_stokes/post_processing/mesh_function_interpolation.h

```diff
@@ -0,0 +1,105 @@
+#ifndef THESIS_POST_PROCESSING_MESH_FUNCTION_INTERPOLATION_H
+#define THESIS_POST_PROCESSING_MESH_FUNCTION_INTERPOLATION_H
+
+/**
+ * @file mesh_function_interpolation.h
+ * @brief Bring mesh functions down in a mesh hierarchy
+ */
+
+#include <lf/base/lf_exception.h>
+#include <lf/refinement/mesh_hierarchy.h>
+
+#include <vector>
+
+namespace projects::ipdg_stokes {
+
+namespace post_processing {
+
+/**
+ * @brief A MeshFunction representing another MeshFunction on a coarser mesh
+ * @tparam INNER_MF The MeshFunction on the coarser mesh
+ */
+template <typename INNER_MF>
+class MeshFunctionInterpolation {
+ public:
+  /**
+   * @brief Create a new MeshFunction on refinement level `to` from a mesh
+   * function on refinement level `from`
+   * @param inner The MeshFunction on level `from`
+   * @param meshes The MeshHierarchy the levels refer to
+   * @param from The refinement level of the coarser mesh
+   * @param to The ferinement level of the finer mesh
+   *
+   * Currently only meshes with affine geometries are supported. This could
+   * change if a function lf::geometry::Geometry::Local(Eigen::MatrixXd &global)
+   * would be provided.
+   */
+  MeshFunctionInterpolation(const INNER_MF &inner,
+                            const lf::refinement::MeshHierarchy &meshes,
+                            lf::base::size_type from, lf::base::size_type to)
+      : inner_(inner),
+        parents_(meshes.getMesh(to)->NumEntities(0)),
+        mesh_to_(meshes.getMesh(to)) {
+    // Initialize a map from entities in the fine mesh to entities in the coarse
```

Comment:
Maybe add an `LF_ASSSERT_MSG(from <= to, "MeshFunctionInterpolation can only interpolate from a coarse mesh to a fine mesh.")`...